lab_1/main.py

```diff
@@ -62,9 +89,19 @@ def compare_profiles(unknown_profile: dict, profile_to_compare: dict, top_n: int
     :param top_n: a number of the most common words
     :return: the distance
     """
+    if isinstance(unknown_profile, dict) and isinstance(profile_to_compare, dict) and isinstance(top_n, int):
+        profile_compared = profile_to_compare['freq']
+        profile_compared_top = get_top_n_words(profile_compared, top_n)
+        unknown_profile_needed = unknown_profile['freq']
+        unknown_profile_top = get_top_n_words(unknown_profile_needed, top_n)
+        common_tokens = 0
+        for i in unknown_profile_top:
+            if i in profile_compared_top:
+                common_tokens += 1
+            distance = round(common_tokens/(len(profile_compared_top)), 2)
```

Comment:
Are you sure about calculating the distance each time of the `for i in unknown_profile_top:` loop?